tests/run-make/inaccessible-temp-dir/rmake.rs

```diff
@@ -0,0 +1,35 @@
+// Issue #66530: We would ICE if someone compiled with `-o /dev/null`,
+// because we would try to generate auxiliary files in `/dev/` (which
+// at least the OS X file system rejects).
+//
+// An attempt to `-Ztemps-dir` into a directory we cannot write into should
+// indeed be an error; but not an ICE.
+//
+// However, some folks run tests as root, which can write `/dev/` and end
+// up clobbering `/dev/null`. Instead we'll use an inaccessible path, which
+// also used to ICE, but even root can't magically write there.
+//
+// Note that `-Ztemps-dir` uses `create_dir_all` so it is not sufficient to
+// use a directory with non-existing parent like `/does-not-exist/output`.
+// See https://github.com/rust-lang/rust/issues/66530
+
+//@ ignore-arm
+// Reason: linker error on `armhf-gnu`
+
+use run_make_support::{fs_wrapper, rustc, test_while_readonly};
+
+fn main() {
+    // Create an inaccessible directory.
+    fs_wrapper::create_dir("inaccessible");
+    test_while_readonly("inaccessible", || {
```

Comment:
I realized that this does not reproduce the original test exactly, because it used `chmod(000)`. The test mostly deals with the writeability of `-Ztemps-dir`, so hopefully it should be the same, but I'm still a bit hesitant. On the other hand, you had now enabled the test also on other OSes, which is a win (and there we cannot use `chmod(000)`).

Let's keep it like this for now. If we wanted to make it more general in the future, we could do something like this:
```rust
with_permissions("inaccessible", |mut perms| {
   perms.set_mode(0o000);
   perms
}, || {
    // Contents of test
});
```

Reply:
Yes, I noticed this as well, and had written the 000 form initially. But, as you said, in order to avoid having to sprinkle `//@ ignore-windows` everywhere, I opted for this instead.